### ISSUE TITLE
Error: spawn npm.cmd ENOENT

### ISSUE THREAD

**Issue by paullam328** [aws-amplify/amplify-cli#7364]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

v14.15.5

### Amplify CLI Version

4.51.1

### What operating system are you using?

Mac

### Amplify Categories

hosting

### Amplify Commands

publish

### Describe the bug

amplify publish fails as spawn npm.cmd.  I am using MacOS.  Almost every related post regarding the issue is for Windows.

```
Pauls-MBP:project paullam$ amplify publish
✔ Successfully pulled backend environment dev from the cloud.

Current Environment: dev

| Category | Resource name            | Operation | Provider plugin   |
| -------- | ------------------------ | --------- | ----------------- |
| Auth     | authapp12345 | No Change | awscloudformation |
| Hosting  | S3AndCloudFront          | No Change | awscloudformation |

No changes detected
frontend build command execution error
Error: spawn npm.cmd ENOENT
    at Process.ChildProcess._handle.onexit (internal/child_process.js:269:19)
    at onErrorNT (internal/child_process.js:465:16)
    at processTicksAndRejections (internal/process/task_queues.js:80:21) {
  errno: -2,
  code: 'ENOENT',
  syscall: 'spawn npm.cmd',
  path: 'npm.cmd',
  spawnargs: [ 'run-script', 'build' ]
}
An error occurred during the publish operation: spawn npm.cmd ENOENT
2021-05-20T02:36:21.466Z - error: uncaughtException: spawn npm.cmd ENOENT date=Wed May 19 2021 19:36:21 GMT-0700 (Pacific Daylight Time), pid=44222, uid=501, gid=20, cwd=/Users/paullam/Desktop/project, execPath=/usr/local/bin/node, version=v14.15.5, argv=[/usr/local/bin/node, /usr/local/bin/amplify, publish], rss=377028608, heapTotal=318607360, heapUsed=270009488, external=9229901, arrayBuffers=7109275, loadavg=[2.66552734375, 2.41650390625, 2.513671875], uptime=602653, trace=[column=19, file=internal/child_process.js, function=Process.ChildProcess._handle.onexit, line=269, method=onexit, native=false, column=16, file=internal/child_process.js, function=onErrorNT, line=465, method=null, native=false, column=21, file=internal/process/task_queues.js, function=processTicksAndRejections, line=80, method=null, native=false], stack=[Error: spawn npm.cmd ENOENT,     at Process.ChildProcess._handle.onexit (internal/child_process.js:269:19),     at onErrorNT (internal/child_process.js:465:16),     at processTicksAndRejections (internal/process/task_queues.js:80:21)]
```

### Expected behavior

By running amplify publish I expect the code to deploy to Amplify correctly.

### Reproduction steps

git clone the application
amplify init
amplify pull
amplify status
amplify publish

Like, other people do the same steps and worked for them, but not for me

### GraphQL schema(s)

<details>

```graphql
# Put schemas below this line


```

</details>

### Log output

<details>

```

```

</details>

### Additional information
```
amplify configure project -l

Project information
| Name: nomadapplication
| Environment: dev
| Default editor: vscode
| App type: javascript
| Javascript framework: react
| Source Directory Path: src
| Distribution Directory Path: build
| Build Command: npm run-script build
| Start Command: npm run-script start

```

**Comment by ammarkarachi** [comment]
@paullam328 What's the output when you run `node -e 'console.log(/^win/.test(process.platform))'`?

**Comment by paullam328** [comment]
Aha, I finally figured out the issue :smile:...  I am using MacOS, and I think npm.cmd run-script build within the Amplify configuration is the culprit.
This Amplify configuration is probably working for the contributor in my project since he is using Windows, and he created this project and configured it using Windows.

Right now, I am using a set of new Amplify configurations: as below, and amplify publish works successfully now!
(The Build Command as shown in the “Project information” section with "amplify configure project -l" is deceiving… I have already changed from npm.cmd run-script build to npm run build, but it’s still displaying “Build Command: npm run-script build”…)

**Comment by paullam328** [comment]
Aha, I finally figured out the issue :smile:...  I am using MacOS, and I think npm.cmd run-script build within the Amplify configuration is the culprit.
This Amplify configuration is probably working for the contributor in my project since he is using Windows, and he created this project and configured it using Windows.

Right now, I am using a set of new Amplify configurations: as below, and amplify publish works successfully now!
(The Build Command as shown in the “Project information” section with "amplify configure project -l" is deceiving… I have already changed from npm.cmd run-script build to npm run build, but it’s still displaying “Build Command: npm run-script build”…)
